### PR TITLE
created another test case for format

### PR DESCRIPTION
**Pull request by agamble33**
Added a test for format method pending error

### CODE REVIEW

**Review by agamble33 — COMMENTED** [review]
Created three different dynamic tests one with three dynamic arguments. One with no arguments and one with two.

File: src/test/javascript/stringExtensionSpec.js [hunk 1]
```diff
@@ -56,6 +56,12 @@ define(['bytepushers'], function(BytePushers) {
 
                 expect(result).toBe(testStr); // testStr = "Hello this is a random string";
             });
+
+            it('can format a string properly', function() {
+                var result = String.format("I'm {1} a {2} powerful {3}", "frontend", "developer");
+
+                expect(result).toBe(testStr); // testStr = "I'm a powerful frontend developer";
```
Comment (ocampd1) [inline src/test/javascript/stringExtensionSpec.js]: Needs actualResult on line in order to reflect with code.

**Review by ocampd1 — CHANGES_REQUESTED** [review]
Needs actualResult on line in order to reflect with code.

File: src/main/javascript/software.bytepushers.string.extensions.js [hunk 2]
```diff
@@ -62,7 +62,7 @@
     /**
      * <p>Convenience function that will format a string with dynamic variables.</p>
      * @static
-     * @param {...string} string - first argument is the string template to be formatted.  The remaining arguments are the format items (e.g. "{0}")
+     * @param {...string} string - first argument is the string template to be formatted.  The remaining arguments are the format/dynamic items (e.g. "{0}")
```
Comment (tonte-pouncil) [inline src/main/javascript/software.bytepushers.string.extensions.js]: Let change the documentation to say: The remaining arguments will be dynamically injected into template variable placeholders.

File: src/test/javascript/stringExtensionSpec.js [hunk 3]
```diff
@@ -1,46 +1,46 @@
...
         describe('String.prototype.toNormalCase', function () {
-            it('can format a sentence into camel case', function() {
+            it('can format a sentence into camel case', function () {
                 var result = "Hello world".toCamelCase().toNormalCase();
                 // helloWorld => "Hello World"
                 expect(result).toBe("Hello World");
             });
 
-            it('can format a sentence into camel case with a single letter word', function() {
+            it('can format a sentence into camel case with a single letter word', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 4]
```diff
@@ -51,13 +51,39 @@ define(['bytepushers'], function(BytePushers) {
         });
 
         describe('String.format', function () {
-            it('can format a string properly', function() {
+            it('can format a string properly', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 5]
```diff
@@ -1,46 +1,46 @@
...
+            it('can format a sentence into camel case', function () {
                 var result = "Hello this is a sentence.".toCamelCase();
 
                 expect(result).toBe("helloThisIsASentence.");
             });
         });
 
         describe('String.prototype.toNormalCase', function () {
-            it('can format a sentence into camel case', function() {
+            it('can format a sentence into camel case', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 6]
```diff
@@ -1,46 +1,46 @@
...
+            it('can tell if a string does not include a string', function () {
                 var result = testStr.includes("This is not in the string");
 
                 expect(result).toBe(false);
             });
         });
 
         describe('String.prototype.toCamelCase', function () {
-            it('can format a sentence into camel case', function() {
+            it('can format a sentence into camel case', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 7]
```diff
@@ -1,46 +1,46 @@
...
         describe('String.prototype.includes', function () {
-            it('can tell if a string includes a string', function() {
+            it('can tell if a string includes a string', function () {
                 var result = testStr.includes("o this i");
 
                 expect(result).toBe(true);
             });
 
-            it('can tell if a string does not include a string', function() {
+            it('can tell if a string does not include a string', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 8]
```diff
@@ -1,46 +1,46 @@
...
+            it('will trim whitespace correctly', function () {
                 var result = "                       l     shdasdk;lg;al j          ".trim();
 
                 expect(result).toBe("l     shdasdk;lg;al j");
             });
         });
 
         describe('String.prototype.includes', function () {
-            it('can tell if a string includes a string', function() {
+            it('can tell if a string includes a string', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 9]
```diff
@@ -1,46 +1,46 @@
 define(['bytepushers'], function(BytePushers) {
-    describe("String extension tests:", function() {
+    describe("String extension tests:", function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 10]
```diff
@@ -1,46 +1,46 @@
...
 
         var testStr = "Hello this is a random string";
 
         describe('String.prototype.trim', function () {
-            it('will trim whitespace correctly', function() {
+            it('will trim whitespace correctly', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 11]
```diff
@@ -51,13 +51,39 @@ define(['bytepushers'], function(BytePushers) {
...
+            it('can format a string properly', function () {
                 var result = String.format("Hello {1} is {2} random {3}", "this", "a", "string");
 
                 expect(result).toBe(testStr); // testStr = "Hello this is a random string";
             });
-        });
-    });
-});
 
+            it('can format two dynamic variables into template.', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 12]
```diff
@@ -51,13 +51,39 @@ define(['bytepushers'], function(BytePushers) {
...
+                var templateUnderTest = "I'm a {1} web {2}.";
+                var expectedResult = "I'm a powerful web developer.";
+                var actualResult = String.format(templateUnderTest, "powerful", "developer");
+                expect(actualResult).toBe(expectedResult); // testStr = "I'm a powerful web developer";
+            });
+            it('can format three dynamic variables into template.', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 13]
```diff
@@ -51,13 +51,39 @@ define(['bytepushers'], function(BytePushers) {
...
+                var templateUnderTest = "I'm a {1} {2} web {3}.";
+                var expectedResult = "I'm a very powerful web developer.";
+                var actualResult = String.format(templateUnderTest, "very", "powerful", "developer");
+                expect(actualResult).toBe(expectedResult); // testStr = "I'm a very powerful web developer";
+            });
+
+            it('can format with no dynamic variables into template.', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

File: src/test/javascript/stringExtensionSpec.js [hunk 14]
```diff
@@ -51,13 +51,39 @@ define(['bytepushers'], function(BytePushers) {
...
+                var templateUnderTest = "I'm a powerful web developer.";
+                var expectedResult = "I'm a powerful web developer.";
+                var actualResult = String.format(templateUnderTest, "I'm a powerful web developer.");
+                expect(actualResult).toBe(expectedResult); // testStr = "I'm a powerful web developer";
 
+            });
+            it('can format one dynamic variables into template.', function () {
```
Comment (tonte-pouncil) [inline src/test/javascript/stringExtensionSpec.js]: Remove space between function and open parentheses.

**Review by tonte-pouncil — CHANGES_REQUESTED** [review]

**Review by tonte-pouncil — APPROVED** [review]

**Pull request merged**